### DISCUSSION TITLE
git pull make a fetch ? so "--ff-only" seems not possible

### DISCUSSION THREAD

**Discussion by fletort** [gitpython-developers/GitPython#1172] [Q&A] [answered]
With this code:
origin.pull('--ff-only')

 I have the following error :

>           raise GitCommandError(self.args, status, errstr)
E           git.exc.GitCommandError: Cmd('git') failed due to: exit code(1)
E             cmdline: git pull -v origin --ff-only master
E             stderr: 'error: unknown option `ff-only master'
E           usage: git fetch [<options>] [<repository> [<refspec>...]]
E              or: git fetch [<options>] <group>
E              or: git fetch --multiple [<options>] [(<repository> | <group>)...]
E              or: git fetch --all [<options>]

In fact we see that the commande used is the fetch one.... not the pull one... so the --ff-only option is not available...
how can i make a pull that make a pull ?

**Comment by Byron** [comment, marked as answer]
After taking a look at [the API reference](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=pull#git.remote.Remote.pull) I believe `origin.pull(ff_only=true)` should do the trick.

Please feel free to keep posting here in case that didn't work.
